packages/contracts-bedrock/src/L1/USDXBridge.sol

```diff
@@ -0,0 +1,171 @@
+
+    /// @notice Contract of the Optimism Portal.
+    /// @custom:network-specific
+    OptimismPortal public portal;
+
+    /// @notice Address of the System Config contract.
+    SystemConfig public config;
+
+    /// @notice Addresses of allow-listed stablecoins.
+    mapping(address => bool) public allowlisted;
+
+    /// @notice The limit to the total USDX supply that can be minted and bridged.
+    uint256 public depositCap;
+
+    /// @notice The total amount of USDX bridged via this contract.
+    uint256 public totalBridged;
+
+    /// INITIALIZE ///
+
+    constructor() {
+        initialize({
+            _owner: address(0xdEaD),
+            _portal: OptimismPortal(payable(address(0))),
+            _config: SystemConfig(payable(address(0))),
+            _stablecoins: new address[](0),
+            _depositCap: 0
+        });
+    }
+
+    /// @notice The initialization function used during the proxy contract set up.
+    /// @param  _owner The address granted ownership rights to this contract.
+    /// @param  _portal The Optimism Portal contract, which is directly responsible for bridging USDX.
+    /// @param  _config The Optimism System Config contract, which ensures alignment on the gas token.
+    /// @param  _stablecoins An array of allow-listed stablecoins that can be used to mint and bridge USDX.
+    /// @param  _depositCap The initial deposit cap for this contract, which limits the total amount bridged.
+    /// @dev    This function includes an unbounded for-loop. Ensure that the array of allow-listed
+    ///         stablecoins is reasonable in length.
+    function initialize(
+        address _owner,
+        OptimismPortal _portal,
+        SystemConfig _config,
+        address[] memory _stablecoins,
+        uint256 _depositCap
+    ) public initializer {
+        /// Set storage
+        __Ownable_init();
+        transferOwnership(_owner);
+        portal = _portal;
+        config = _config;
+        depositCap = _depositCap;
+        /// Add allow-listed stablecoins for proxy set up
+        if (address(config) != address(0)) {
+            uint256 length = _stablecoins.length;
+            for (uint256 i; i < length; ++i) {
+                allowlisted[_stablecoins[i]] = true;
+            }
+        }
+    }
+
+    /// BRIDGE ///
+
+    /// @notice This function allows users to deposit any allow-listed stablecoin to the Ozean Layer L2.
+    /// @param  _stablecoin Depositing stablecoin address.
+    /// @param  _amount The amount of deposit stablecoin to be swapped for USDX.
+    /// @param  _to Recieving address on L2.
+    function bridge(
+        address _stablecoin,
+        uint256 _amount,
+        address _to
+    ) external nonReentrant {
+        /// Checks
+        require(allowlisted[_stablecoin], "USDXBridge: Stablecoin not accepted.");
+        require(_amount > 0, "USDXBridge: May not bridge nothing.");
+        uint256 bridgeAmount = _getBridgeAmount(_stablecoin, _amount);
```

Comment:
Good check here for conversion between 6 decimal places for USDT / USDC and USDX should be same.